test/src/org/labkey/test/tests/targetedms/TargetedMSTrailingMeanAndCVTest.java

```diff
@@ -0,0 +1,177 @@
+import org.labkey.test.components.targetedms.QCPlotsWebPart;
+import org.labkey.test.pages.targetedms.PanoramaDashboard;
+import org.labkey.test.util.PortalHelper;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+@Category({})
+@BaseWebDriverTest.ClassTimeout(minutes = 5)
+public class TargetedMSTrailingMeanAndCVTest extends TargetedMSTest
+{
+    @BeforeClass
+    public static void initProject()
+    {
+        TargetedMSTrailingMeanAndCVTest init = (TargetedMSTrailingMeanAndCVTest) getCurrentTest();
+        init.doInit();
+    }
+
+    private void doInit()
+    {
+        setupFolder(FolderType.QC);
+        importData(SProCoP_FILE);
+
+        goToProjectHome();
+        PortalHelper portalHelper = new PortalHelper(this);
+        portalHelper.moveWebPart("QC Plots", PortalHelper.Direction.UP);
+    }
+
+    @Override
+    protected String getProjectName()
+    {
+        return getClass().getSimpleName() + " Project";
+    }
+
+    @Before
+    public void deletingGuideSet()
+    {
+        goToProjectHome();
+        removeAllGuideSets();
+    }
+
+    @Test
+    public void testTrailingMeanPlotType()
+    {
+        goToProjectHome();
+        PanoramaDashboard dashboard = new PanoramaDashboard(this);
+        QCPlotsWebPart qcPlotsWebPart = dashboard.getQcPlotsWebPart();
+        qcPlotsWebPart.setQCPlotTypes(QCPlotsWebPart.QCPlotType.TrailingMean);
+
+        log("Verifying error checking on trailing last number");
+        qcPlotsWebPart.setTrailingLast("-1");
+        longWait().until(ExpectedConditions.textToBePresentInElement(Locators.labkeyError.findWhenNeeded(getDriver()),
+                "TrailingMean - Please enter a positive integer (>2) that is less than or equal to total number of available runs - 47"));
+
+        qcPlotsWebPart.setTrailingLast("48");
+        longWait().until(ExpectedConditions.textToBePresentInElement(Locators.labkeyError.findWhenNeeded(getDriver()),
+                "TrailingMean - The number you entered is larger than the number of available runs. Only 47 runs are used for calculation"));
+
+        log("Verifying y-axis value based on metric type");
+        qcPlotsWebPart.setTrailingLast("3");
+        qcPlotsWebPart.setMetricType(QCPlotsWebPart.MetricType.MASSACCURACY);
+        Assert.assertTrue("Y axis is not labeled correctly for " + QCPlotsWebPart.MetricType.MASSACCURACY,
+                qcPlotsWebPart.getSVGPlotText("precursorPlot0").contains("PPM"));
+
+        qcPlotsWebPart.setMetricType(QCPlotsWebPart.MetricType.RETENTION);
+        Assert.assertTrue("Y axis is not labeled correctly for " + QCPlotsWebPart.MetricType.RETENTION,
+                qcPlotsWebPart.getSVGPlotText("precursorPlot0").contains("Minutes"));
+
+        log("Verifying the count of points on the plot");
+        Assert.assertEquals("Invalid point count for all replicates", 315 , qcPlotsWebPart.getPointElements("d", SvgShapes.CIRCLE.getPathPrefix(), true).size());
```

Comment:
This should explain how 315 is calculated. This sort of assertion can be tricky to reverse-engineer if there is ever a failure.
e.g. `7 plots with 45 points per plot`
Better yet, all of these values should be constants and variables that can be used to calculate expectations, making only once spot to update if the underlying data changes.
e.g. `int expected = REPLICATE_COUNT * (RUN_COUNT - trailingLast + 1)`

The same goes for the other row count assertions.

Reply:
what is RUN_COUNT?

Reply:
Those were just hypothetical constants.
e.g.
```
private static final RUN_COUNT = 47;
private static final REPLICATE_COUNT = 7;
```